bindings/node.js/kzg.cxx

```diff
@@ -32,41 +33,67 @@ Napi::Value throw_invalid_argument_type(const Napi::Env env, std::string name, s
+ * - get_blob
+ * - get_commitment
+ * - get_proof
+ * - get_bytes_32
+ * 
+ * Built to pass in a raw Napi::Value so it can be used like
+ * `get_bytes(env, info[0])` or can also be used to pull props from
+ * arrays like `get_bytes(env, passed_napi_array[2])`
+ * 
+ * Designed to raise the correct javascript exception and return a
+ * valid pointer to the calling context to avoid native stack-frame
+ * unwinds.  Calling context can check for `nullptr` to see if an
+ * exception was raised or a valid pointer was returned from V8
+ * 
+ * @param[in] env - napi_env passed from calling context
+ * @param[in] val - napi_value
+ * @param[in] length - size_t to validate ArrayBuffer against
+ * @param[in] name - name of prop being validated for error reporting
+ * 
+ * @return - native pointer to first byte in ArrayBuffer
+ */
+inline uint8_t *get_bytes(
+    const Napi::Env &env,
+    const Napi::Value &val,
+    size_t length,
+    std::string_view name)
+{
+  if (!val.IsTypedArray() || val.As<Napi::TypedArray>().TypedArrayType() != napi_uint8_array) {
+    std::ostringstream msg;
+    msg << "Expected " << name << " to be a Uint8Array";
+    Napi::TypeError::New(env, msg.str()).ThrowAsJavaScriptException();
+    return nullptr;
   }
-
-  // Wrap up the std::vector into the ArrayBuffer.
-  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
-      env,
-      vector->data(),
-      length /* size in bytes */,
-      [](Napi::Env /*env*/, void* /*data*/, std::vector<uint8_t>* hint) {
-        std::unique_ptr<std::vector<uint8_t>> vectorPtrToDelete(hint);
-      },
-      vector.get());
-
-  // The finalizer is responsible for deleting the vector: release the
-  // unique_ptr ownership.
-  vector.release();
-
-  return Napi::Uint8Array::New(env, length, buffer, 0);
-}
-
-const uint8_t * extract_byte_array_from_param(const Napi::CallbackInfo& info, const int index, const std::string name) {
-  auto param = info[index].As<Napi::TypedArray>();
-  if (!param.IsTypedArray() || param.TypedArrayType() != napi_uint8_array) {
-    throw_invalid_argument_type(info.Env(), name, "UInt8Array");
+  Napi::Uint8Array array = val.As<Napi::Uint8Array>();
+  if (array.ByteLength() != length) {
+    std::ostringstream msg;
+    msg << "Expected " << name << " to be " << length << " bytes";
+    Napi::TypeError::New(env, msg.str()).ThrowAsJavaScriptException();
+    return nullptr;
   }
-  return param.As<Napi::Uint8Array>().Data();
+  return array.Data();
+}
+inline Blob *get_blob(const Napi::Env &env, const Napi::Value &val) {
+  return reinterpret_cast<Blob *>(get_bytes(env, val, BYTES_PER_BLOB, "blob"));
+}
+inline KZGCommitment *get_commitment(const Napi::Env &env, const Napi::Value &val) {
+  return reinterpret_cast<KZGCommitment *>(get_bytes(env, val, BYTES_PER_COMMITMENT, "commitmentBytes"));
+}
+inline KZGProof *get_proof(const Napi::Env &env, const Napi::Value &val) {
+  return reinterpret_cast<KZGProof *>(get_bytes(env, val, BYTES_PER_PROOF, "proofBytes"));
```

Comment:
This was a bad suggestion from my side, but I think it's better if instead of `get_proof()` and `get_commitment()` we just have `get_bytes_48()` for parsing.

There is a convention we are using in the code and spec, where untrusted bytes (before they get into the KZG library) are called `Bytes32` or `Bytes48` and they only get turned to `KZGProof` and `KZGCommitment` after they've been validated and normalized by the library.

For this reason, and because the bindings are receiving untrusted bytes as input (from the perspective of the KZG library) I think we should handle them `Bytes48` when they are passed into the lib. OTOH, it's fine for the KZG library to return a `KZGProof`.

Reply:
Yeah good call, I agree. Also please do `get_bytes48` (without the second underscore).

Reply:
Ok.  Updated so that the only places that `KZGCommitment` and `KZGProof` exist now are as returns of `BlobToKzgCommitment`, `ComputeKzgProof` and `ComputeBlobKzgProof`.  Everywhere else the code refers to `Bytes48`.  

I also updated both names to not have the underscore preceding the number